Content.Shared/Eye/Blinding/Systems/BlindableSystem.cs

```diff
@@ -62,13 +62,31 @@ public void AdjustEyeDamage(Entity<BlindableComponent?> blindable, int amount)
             return;
 
         blindable.Comp.EyeDamage += amount;
-        blindable.Comp.EyeDamage = Math.Clamp(blindable.Comp.EyeDamage, 0, BlindableComponent.MaxDamage);
+        UpdateEyeDamage(blindable, true);
+    }
+    public void UpdateEyeDamage(Entity<BlindableComponent?> blindable, bool isDamageChanged)
+    {
+        if (!Resolve(blindable, ref blindable.Comp, false))
+            return;
+
+        var previousDamage = blindable.Comp.EyeDamage;
+        blindable.Comp.EyeDamage = Math.Clamp(blindable.Comp.EyeDamage, blindable.Comp.MinDamage, BlindableComponent.MaxDamage);
         Dirty(blindable);
-        UpdateIsBlind(blindable);
+        if (!isDamageChanged && previousDamage == blindable.Comp.EyeDamage)
+            return;
 
+        UpdateIsBlind(blindable);
         var ev = new EyeDamageChangedEvent(blindable.Comp.EyeDamage);
         RaiseLocalEvent(blindable.Owner, ref ev);
     }
+    public void SetMinDamage(Entity<BlindableComponent?> blindable, int amount)
+    {
+        if (!Resolve(blindable, ref blindable.Comp, false))
+            return;
+
+        blindable.Comp.MinDamage = amount;
+        UpdateEyeDamage(blindable, false);
+    }
 }
```

Comment:
There's no reason to make UpdateEyeDamge public, additionally setting the min / max should be clamping it and not just clamping if the max changes.